### DISCUSSION TITLE
Combine parts into one layer with different z-index

### DISCUSSION THREAD

**Discussion by nexxus-zz** [nftchef/art-engine#34] [Help + Q&A] [answered]
Hey, 
thanks for supporting this lib first of all! 

I have a question if one thing is possible with the library or not: 
Imagine I have a trait that needs to have different layers attached to it (in my case 2), 
So one trait has the left part on level 3 and the right part on level 5 let's say. 
if it would be one trait (without left and right part) I would just add them into `layerConfigurations` and all would be fine.
With this trait being split I thought I would use one folder and have the 2 traits inside it (so using "Required Files" from the readme). But using this I seem to not be able to combine it with controlling the z-index.
So what I tried: 
Create a folder with `Trait1#80`
Inside this folder, I would have: 
```
1-PropertyLeft.png
2-PropertyRight.png
```

What I would like to have: 
```
1-z+0,PropertyLeft.png
2-z+2,PropertyRight.png
```
or something like that.

Can you think of another way to combine required files (splitting left and right part of the image, but using as one trait) and setting a z-index per property? 

Thanks a lot in advance!

**Comment by nftchef** [comment, marked as answer]
Hi! you're on the right track, this is exactly how it is designed to be used.

typically, you would have the trait layered in the "proper" order for the main piece of the trait or the majority of the traits. 

Using hair for example, if you have a hair trait with one piece that needs to go behind another layer, you would do as you've outlined

```
|--Hair/
|---- HairTrait#50/
|----  front.png
|---- z-1,back.png
```

The difference from your example is subtle, but basically, you are avoiding setting the `z#` on every file and only moving the ones that are necessary by first using `layersOrder` to organize staking. Also, in your example `z+` is invalid, positive integers do not need a sign, `z2,` for example.

**Reply by nexxus-zz** [reply]
Awesome, thanks for the answer, it worked like a charm!
So close indeed :D
